src/ui/libs/DatalensChartkit/modules/data-provider/charts/ui-sandbox-runtime.ts

```diff
@@ -0,0 +1,202 @@
+                    items.push(objHandle);
+
+                    setItem(value, objHandle);
+                } else if (typeof value === 'function') {
+                    const fnHandle = this.vm.newFunction(key, (...fnArgs) => {
+                        let callEnded = false;
+                        const nativeArgs = fnArgs.map((fnArg) => {
+                            const result = this.vm.dump(fnArg);
+
+                            if (isFunction(result)) {
+                                const longLivedCallbackHandle = fnArg.dup();
+                                this.pending.push(longLivedCallbackHandle);
+
+                                return (...args: unknown[]) => {
+                                    const fnContext = this.vm.newObject();
+                                    const mappedArgs = args.map(
+                                        (a) => this.toHandle(a) as QuickJSHandle,
+                                    );
+                                    const fnResult = this.vm.callFunction(
+                                        longLivedCallbackHandle,
+                                        fnContext,
+                                        ...mappedArgs,
+                                    );
+                                    const fnResult2 = this.getFunctionResult(fnResult);
+                                    fnContext.dispose();
+                                    mappedArgs.forEach((arg) => arg.dispose());
+
+                                    setTimeout(() => {
+                                        if (callEnded) {
+                                            if (longLivedCallbackHandle.alive) {
+                                                longLivedCallbackHandle.dispose();
+                                            }
+                                            this.dispose();
+                                        }
+                                    }, 0);
+
+                                    return fnResult2;
+                                };
+                            }
+
+                            return result;
+                        });
+                        const result = value(...nativeArgs);
+                        callEnded = true;
+                        return this.toHandle(result);
+                    });
+                    this.vm.setProp(parent, key, fnHandle);
+                    items.push(fnHandle);
+                }
+            });
+        };
+
+        setItem(api, this.vm.global);
+
+        items.forEach((item) => item.dispose());
+    }
+
+    private getFunctionResult(result: VmCallResult<QuickJSHandle>) {
+        if (result.error) {
+            const errorMsg = this.vm.dump(result.error);
+            result.error.dispose();
+            throw errorMsg;
+        }
+
+        const value = this.vm.dump(result.value);
+        result.value.dispose();
+
+        return value;
+    }
+
+    private toHandle(value: unknown) {
```

Comment:
You can add return type `QuickJSHandle` and then you won't need in cast like `this.toHandle(a) as QuickJSHandle`